app/src/main/java/com/infomaniak/drive/ui/MainViewModel.kt

```diff
@@ -343,4 +349,51 @@ class MainViewModel(appContext: Application) : AndroidViewModel(appContext) {
             UploadFile.deleteAll(fileDeleted)
         }
     }
+
+    fun getRecentChanges(driveId: Int, forceDownload: Boolean = false): LiveData<FileListFragment.FolderFilesResult?> {
+        getRecentChangesJob.cancel()
+        getRecentChangesJob = Job()
+
+        val ignoreDownload = lastModifiedTime != 0L && (Date().time - lastModifiedTime) < DOWNLOAD_INTERVAL && !forceDownload
+
+        return liveData(Dispatchers.IO + getRecentChangesJob) {
+            if (ignoreDownload) {
+                emit(
+                    FileListFragment.FolderFilesResult(
+                        files = recentlyChangedFiles.data ?: arrayListOf(),
+                        isComplete = true,
+                        page = 1
+                    )
+                )
+                return@liveData
+            }
+
+            suspend fun recursive(page: Int) {
+                val apiResponse = ApiRepository.getLastModifiedFiles(driveId, page)
+                if (apiResponse.isSuccess()) {
+                    val data = apiResponse.data
+                    data?.let { FileController.storeRecentChanges(it) }
```

Comment:
We would have to adapt storeRecentChanges so that we only reset on the 1st page

Reply:
Code has been adjusted 👍